### PR TITLE
[MOS-636] Added MMI/USSD confirmation

### PR DESCRIPTION
**Pull request by kkleczkowski**
USSD/MMI codes are now confirmed after they are sended.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible
- [x] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

File: module-services/service-cellular/service-cellular/requests/UssdRequest.hpp [hunk 1]
```diff
@@ -12,10 +12,10 @@ namespace cellular
 {
     constexpr inline auto UssdRegex = "^[\\*\\#].*[\\#]$";
 
-    class UssdRequest : public Request
+    class UssdRequest : public ConfirmingRequest
     {
       public:
-        UssdRequest(const std::string &data) : Request(data){};
+        UssdRequest(const std::string &data) : ConfirmingRequest(data){};
```
Comment (SP2FET) [inline module-services/service-cellular/service-cellular/requests/UssdRequest.hpp:18]:
```suggestion
        explicit UssdRequest(const std::string &data) : ConfirmingRequest(data){};
```

**Review by SP2FET — APPROVED** [review]

**Comment by SP2FET** [comment]
before merging - please correct typos in the PR and commit message ;)

File: module-apps/application-desktop/windows/MmiConfirmationWindow.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+#include "MmiConfirmationWindow.hpp"
+
+#include <i18n/i18n.hpp>
+#include <Image.hpp>
+#include <apps-common/ApplicationCommon.hpp>
+#include <service-appmgr/data/MmiActionsParams.hpp>
+
+using namespace gui;
+
+// move to style
```
Comment (swichu91) [inline module-apps/application-desktop/windows/MmiConfirmationWindow.cpp:13]: Remove it?

File: module-apps/application-desktop/windows/MmiConfirmationWindow.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    style::window::default_body_width,
+                    style::window::default_body_height,
+                    "progress_128px_W_G",
+                    utils::translate("app_desktop_info_mmi_confirmation"));
+    icon->setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Top));
+    icon->image->setMargins(Margins(0, icon::image_top_margin, 0, icon::image_bottom_margin));
+}
+
+void MmiConfirmationWindow::onBeforeShow(ShowMode mode, SwitchData *data)
+{}
```
Comment (swichu91) [inline module-apps/application-desktop/windows/MmiConfirmationWindow.cpp:48]: If you don't need it then don't declare it :)

File: module-services/service-cellular/ServiceCellular.cpp [hunk 4]
```diff
@@ -1793,12 +1793,17 @@ auto ServiceCellular::handleCellularCallRequestMessage(cellular::CallRequestMess
 
     auto request = factory.create();
     CellularRequestHandler handler(*this);
+    auto confirmation = request->confirmRequest();
```
Comment (swichu91) [inline module-services/service-cellular/ServiceCellular.cpp:1796]:
```suggestion
    const auto confirmation = request->confirmRequest();
```

File: module-services/service-cellular/requests/Request.cpp [hunk 5]
```diff
@@ -54,4 +59,9 @@ namespace cellular
 
         return cmd;
     }
+
+    std::shared_ptr<sys::DataMessage> ConfirmingRequest::confirmRequest() const noexcept
+    {
+        return std::move(std::make_shared<cellular::MMIConfirmationMessage>());
```
Comment (swichu91) [inline module-services/service-cellular/requests/Request.cpp:65]:
```suggestion
        return std::make_shared<cellular::MMIConfirmationMessage>();
```

File: module-services/service-cellular/service-cellular/CellularMessage.hpp [hunk 6]
```diff
@@ -433,6 +433,19 @@ namespace cellular
         }
     };
 
+    class MMIConfirmationMessage : public sys::DataMessage, public app::manager::actions::ConvertibleToAction
+    {
+      public:
+        explicit MMIConfirmationMessage()
```
Comment (swichu91) [inline module-services/service-cellular/service-cellular/CellularMessage.hpp:439]:
```suggestion
       MMIConfirmationMessage()
```

File: module-services/service-cellular/src/ServiceCellularPriv.cpp [hunk 7]
```diff
@@ -588,7 +588,12 @@ namespace cellular::internal
             channel->cmd(at::AT::SMS_GSM);
             std::string command = at::factory(at::AT::CUSD_SEND) + request + commandDcs;
             auto result         = channel->cmd(command, std::chrono::seconds(120));
```
Comment (swichu91) [inline module-services/service-cellular/src/ServiceCellularPriv.cpp:590]:
```suggestion
            const auto result         = channel->cmd(command, std::chrono::seconds(120));
```

File: module-services/service-cellular/src/ServiceCellularPriv.cpp [hunk 8]
```diff
@@ -588,7 +588,12 @@ namespace cellular::internal
...
-            return result.code == at::Result::Code::OK;
+            if (result.code == at::Result::Code::OK) {
+                owner->bus.sendUnicast(std::move(std::make_shared<cellular::MMIConfirmationMessage>()),
```
Comment (swichu91) [inline module-services/service-cellular/src/ServiceCellularPriv.cpp:592]:
```suggestion
                owner->bus.sendUnicast(std::make_shared<cellular::MMIConfirmationMessage>(),
```

File: module-services/service-cellular/service-cellular/requests/Request.hpp [hunk 9]
```diff
@@ -50,4 +53,11 @@ namespace cellular
         bool isRequestHandled = false;
         std::string request;
     };
+
+    class ConfirmingRequest : public Request
+    {
+      public:
+        ConfirmingRequest(const std::string &data) : Request(data){};
```
[the remark this answers is not in the document]
Comment (kkleczkowski) [reply module-services/service-cellular/service-cellular/requests/Request.hpp:60]: done

File: module-services/service-cellular/service-cellular/requests/UssdRequest.hpp (hunk 1 above)
Comment (kkleczkowski) [reply module-services/service-cellular/service-cellular/requests/UssdRequest.hpp:18]: done

File: module-apps/application-desktop/windows/MmiConfirmationWindow.cpp (hunk 2 above)
Comment (kkleczkowski) [reply module-apps/application-desktop/windows/MmiConfirmationWindow.cpp:13]: done

File: module-apps/application-desktop/windows/MmiConfirmationWindow.cpp (hunk 3 above)
Comment (kkleczkowski) [reply module-apps/application-desktop/windows/MmiConfirmationWindow.cpp:48]: done

File: module-services/service-cellular/ServiceCellular.cpp (hunk 4 above)
Comment (kkleczkowski) [reply module-services/service-cellular/ServiceCellular.cpp:1796]: done

File: module-services/service-cellular/requests/Request.cpp (hunk 5 above)
Comment (kkleczkowski) [reply module-services/service-cellular/requests/Request.cpp:65]: done

File: module-services/service-cellular/service-cellular/CellularMessage.hpp (hunk 6 above)
Comment (kkleczkowski) [reply module-services/service-cellular/service-cellular/CellularMessage.hpp:439]: done

File: module-services/service-cellular/src/ServiceCellularPriv.cpp (hunk 7 above)
Comment (kkleczkowski) [reply module-services/service-cellular/src/ServiceCellularPriv.cpp:590]: done

File: module-services/service-cellular/src/ServiceCellularPriv.cpp (hunk 8 above)
Comment (kkleczkowski) [reply module-services/service-cellular/src/ServiceCellularPriv.cpp:592]: done

File: image/system_a/data/lang/Polski.json [hunk 10]
```diff
@@ -252,6 +252,7 @@
   "app_desktop_info_mmi_imei": "IMEI (MEID)",
   "app_desktop_info_mmi_result_success": "Udało się",
   "app_desktop_info_mmi_result_failed": "Nie udało się",
+  "app_desktop_info_mmi_confirmation": "<text>Please wait...<br></br> Processing USSD code.</text>",
```
Comment (Maciej-Mudita) [inline image/system_a/data/lang/Polski.json:255]: can't be translated into Polish? :D

**Review by Maciej-Mudita — APPROVED** [review]

**Pull request merged**